synapse/lib/ast.py

```diff
@@ -4913,10 +4914,12 @@ async def genr():
                 subr.funcscope = True
                 try:
                     if self.hasemit:
+                        await asyncio.sleep(0)
                         async with contextlib.aclosing(await subr.emitter()) as agen:
                             async for item in agen:
```

Comment:
should we yield for each item emitted from an emitter?

Reply:
We probably **should**. I can't find instances of us yielding currently. We'd be relying on the consumer to be doing something that would yield at this point.